ui/src/components/directory/ParentHeader.vue

```diff
@@ -43,9 +47,34 @@ import ArrayObjectNameTagWithLabel from "@/components/shared/generics/ArrayObjec
 import ActionButtons from "@/components/shared/ActionButtons.vue";
 import TextWithLabel from "@/components/shared/generics/TextWithLabel.vue";
 import { IM, RDF } from "@im-library/vocabulary";
-import { getColourFromType, getFAIconFromType } from "@im-library/helpers/ConceptTypeMethods";
+import { getColourFromType, getFAIconFromType, isQuery, isValueSet } from "@im-library/helpers/ConceptTypeMethods";
+import { computed, Ref, watch, ref, onMounted } from "vue";
+import { useStore } from "vuex";
+import { EntityService } from "@/services";
+import { isObjectHasKeys } from "@im-library/helpers/DataTypeCheckers";
 
+const store = useStore();
 const props = defineProps({ concept: { type: Object as any, required: true } });
+const conceptIri = computed(() => store.state.conceptIri);
+const hasQueryDefinition: Ref<boolean> = ref(false);
+
+onMounted(async () => {
+  hasQueryDefinition.value = await getHasQueryDefinition();
+});
+
+watch(
+  () => conceptIri.value,
```

Comment:
No need for arrow function to watch a reactive (inc. computed). watch(conceptIri, () => {}) should work instead, and would be a deep watcher if it was an object.